tfmesos/scheduler.py

```diff
@@ -181,14 +183,14 @@ def __init__(self, task_spec, role=None, master=None, name=None,
         self.protocol = protocol
         self.forward_addresses = forward_addresses
         self.role = role or '*'
-        self.tasks = []
+        self.tasks = {}
+        self.task_failure_count = {}
         self.job_finished = {}
         for job in task_spec:
             self.job_finished[job.name] = 0
             for task_index in range(job.start, job.num):
-                mesos_task_id = len(self.tasks)
-                self.tasks.append(
-                    Task(
+                mesos_task_id = str(uuid.uuid4())
```

Comment:
@treadstone90 how about add a '<framework-name>-' prefix to the `mesos_task_id`?